From 3b6efd5557807de97d9ef729646990ff58507d98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <P@draigBrady.com>
Date: Wed, 12 Nov 2025 13:25:26 +0000
Subject: [PATCH] tests: env --argv0: fix false failure with symlinked single
 binary

* tests/env/env.sh: Always pass --coreutils-prog=true so that the
program to run can be determined with --enable-single-binary=symlinks.
Also actually verify the expected verbose output.
Reported by the Alpine Linux project.
---
 tests/env/env.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/env/env.sh b/tests/env/env.sh
index c838a49a2..a76b40132 100755
--- a/tests/env/env.sh
+++ b/tests/env/env.sh
@@ -165,12 +165,14 @@ test "$exp" = "$got" || fail=1
 
 # Verify argv0 overriding
 for arg in 'argv0' ''; do
-env -v -a short --argv0=$arg true 2>err || fail=1
+env -v -a short --argv0=$arg true --coreutils-prog=true 2>err || fail=1
 cat <<EOF >err_exp || framework_failure_
 argv0:     '$arg'
 executing: true
    arg[0]= '$arg'
+   arg[1]= '--coreutils-prog=true'
 EOF
+compare err_exp err || fail=1
 done
 
 Exit $fail
-- 
2.51.2

